R/utils.R

```diff
@@ -94,4 +94,74 @@ matchArg <- function(arg, choices) {
 setroworder <- function(x, neworder) {
     .Call(data.table:::Creorder, x, as.integer(neworder), PACKAGE = "data.table")
     invisible(x)
+}
+
+
+#' Which values are In the Bin?
+#' 
+#' whichInBin finds array values that fall within a given bin
+#' @param values numeric vector
+#' @param binString a string that defines a numeric bin. For example '[1, 2)'
+#' @export
+whichInBin <- function(values, binString) {
+  binStart <- findBinStart(binString)
+  binEnd <- findBinEnd(binString)
+
+  inBin <- (values >= binStart) * (values < binEnd)
+  return(inBin)
+}
+
+
+#' Find Bin Start
+#' 
+#' findBinStart finds the start value of a given bin.
+#' @param x string that represents a bin. Ex: '[1, 2)'
+#' @param addTimeZone boolean determines if time zone should be added to dates. Default is FALSE.
+#' @export
+findBinStart <- function(x, addTimeZone = c(FALSE, TRUE)) {
```

Comment:
feel like this should be called findBinStartFromString or something

Reply:
im also curious why we need this? can the frontend not pass the exact bin start and ends? why are we getting a string?